plugin/src/main/kotlin/nl/avisi/anowalke/toolWindow/FluffyListener.kt

```diff
@@ -42,42 +47,89 @@ class FluffyListener(val project: Project) : ToolWindowManagerListener {
-//            log.warn("And now pretty")
-//            expressions?.sanatise()?.forEach { log.warn(it) }
-            return expressions
-        } else {
-            file.children.forEach { getJavaFiles(it) }
+            log.warn("File: ${file.name} is not a directory nor a hidden file.")
+            var expressions = PsiManager.getInstance(project).findFile(file)?.let { psiFile ->
+                psiService.extract(psiFile)
+            }
+            if (expressions != null) {
+                log.warn("Extract function retrieved ${expressions.size} expressions.")
+                expressions = sanitiser.sanitise(expressions)
+            }
+            log.warn("Sanitise function retrieved ${expressions?.size} expressions.")
+            return expressions?.map { expression ->
+                Identifier(file.name, expression)
+            }
+        } else if (file.isDirectory) {
+            log.warn("File: ${file.name} is a directory. Check children for java files.")
+            return file.children.flatMap { getJavaFiles(it).orEmpty() }
         }
         return null
     }
 
     private fun getFilesAsIdentifier(file: VirtualFile): List<Identifier>? {
-        if (file.isDirectory) {
-            // Zoek naar files in de java map ipv files met filetype java, omdat hij sommige filetypes door elkaar haalt
-            if (file.name == "java") {
-                return getJavaFiles(file)?.map { expression ->
-                    Identifier(file.name, expression)
-                }
-            }
+        log.warn("Initiate getFilesAsIdentifier function.")
+        // Zoek naar files in de java map ipv files met filetype java, omdat hij sommige filetypes door elkaar haalt
+        if (file.isDirectory && file.name == "java") {
+            log.warn("Current file: '${file.name}' is a java directory.")
+            val identifiers = getJavaFiles(file)
+            log.warn("getJavaFiles function retrieved ${identifiers?.size} identifiers.")
+            return identifiers
+        } else if (file.isDirectory && file.name != "java") {
+            log.warn("Current file: ${file.name} is not a java directory. Check Children for a directory named 'java'.")
+            return file.children.flatMap { getFilesAsIdentifier(it).orEmpty() }
         } else {
-            file.children.forEach { getFilesAsIdentifier(it) }
+            log.warn("Current file: ${file.name} is not a directory. Ignore file.")
         }
         return null
     }
 
-    private fun scanProject() {
+    private fun scanProject(): List<Identifier> {
+        log.warn("Initiate scanProject function.")
         val identifiers = ProjectRootManager.getInstance(project).contentSourceRoots.flatMap {
             getFilesAsIdentifier(it).orEmpty()
         }
+        identifiers.forEach { log.warn("Expression: ${it.expression}") }.also { log.warn("Aloha") }
```

Comment:
Aloha nog nodig? Is wel vrolijk :)

Reply:
Ahahaha ja hij gaf telkens een lege lijst terug dus dat was om te zien of hij het uberhaupt wel uitvoerde haha.